packages/hacl-star/hacl-star.0.4.0/opam

```diff
@@ -0,0 +1,35 @@
+opam-version: "2.0"
+name: "hacl-star"
+version: "0.4.0"
+maintainer: "Victor Dumitrescu <victor.dumitrescu@nomadic-labs.com>"
+authors: [ "Project Everest" ]
+homepage: "https://hacl-star.github.io/"
+bug-reports: "https://github.com/project-everest/hacl-star/issues"
+dev-repo: "git+https://github.com/project-everest/hacl-star.git"
+license: "Apache-2.0"
+synopsis: "OCaml API for EverCrypt/HACL*"
+depends: [
+  "ocaml" { >= "4.05.0" }
+  "dune" {>= "1.2"}
+  "hacl-star-raw" {= version}
+  "zarith"
+  "cppo" {build}
+  "odoc" {with-doc}
+]
+build: [
+  [
+    "dune" "build" "-p" name "-j" jobs
+    "@doc" {with-doc}
+  ]
+]
+run-test: [
```

Comment:
```suggestion
available: [
  arch != "arm32"
]
run-test: [
```

Reply:
I think we need to backport this change also to 0.3.2: https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/0f49b00fd27084edc9c12a8ce8e84171615d719e

```

#=== ERROR while compiling hacl-star-raw.0.3.2 ================================#
# context              2.1.0 | linux/arm32 | ocaml-base-compiler.4.12.0 | file:///src
# path                 ~/.opam/4.12/.opam-switch/build/hacl-star-raw.0.3.2
# command              ~/.opam/opam-init/hooks/sandbox.sh build make -C raw
# exit-code            2
# env-file             ~/.opam/log/hacl-star-raw-24-65423f.env
# output-file          ~/.opam/log/hacl-star-raw-24-65423f.out
### output ###
# make: Entering directory '/home/opam/.opam/4.12/.opam-switch/build/hacl-star-raw.0.3.2/raw'
# cc -fPIC -I. -I ../kremlin/include -I ../kremlin/kremlib/dist/minimal -Wall -Wextra -Werror -std=c11 -Wno-unused-variable -Wno-unknown-warning-option -Wno-unused-but-set-variable -Wno-unused-parameter -Wno-infinite-recursion -g -fwrapv -D_BSD_SOURCE -D_DEFAULT_SOURCE -fPIC -Wno-unused -Wno-parentheses -Wno-deprecated-declarations -Wno-#warnings -Wno-error=cpp -Wno-cpp -g -std=gnu11 -O3 -I "/home/opam/.opam/4.12/lib/ctypes" -I "/home/opam/.opam/4.12/lib/ocaml"   -c -o Hacl_Spec.o Hacl_Spec.c
# cc -fPIC -I. -I ../kremlin/include -I ../kremlin/kremlib/dist/minimal -Wall -Wextra -Werror -std=c11 -Wno-unused-variable -Wno-unknown-warning-option -Wno-unused-but-set-variable -Wno-unused-parameter -Wno-infinite-recursion -g -fwrapv -D_BSD_SOURCE -D_DEFAULT_SOURCE -fPIC -Wno-unused -Wno-parentheses -Wno-deprecated-declarations -Wno-#warnings -Wno-error=cpp -Wno-cpp -g -std=gnu11 -O3 -I "/home/opam/.opam/4.12/lib/ctypes" -I "/home/opam/.opam/4.12/lib/ocaml"   -c -o Hacl_Kremlib.o Hacl_Kremlib.c
# cc -fPIC -I. -I ../kremlin/include -I ../kremlin/kremlib/dist/minimal -Wall -Wextra -Werror -std=c11 -Wno-unused-variable -Wno-unknown-warning-option -Wno-unused-but-set-variable -Wno-unused-parameter -Wno-infinite-recursion -g -fwrapv -D_BSD_SOURCE -D_DEFAULT_SOURCE -fPIC -Wno-unused -Wno-parentheses -Wno-deprecated-declarations -Wno-#warnings -Wno-error=cpp -Wno-cpp -g -std=gnu11 -O3 -I "/home/opam/.opam/4.12/lib/ctypes" -I "/home/opam/.opam/4.12/lib/ocaml"   -c -o Hacl_Blake2s_32.o Hacl_Blake2s_32.c
# cc -fPIC -I. -I ../kremlin/include -I ../kremlin/kremlib/dist/minimal -Wall -Wextra -Werror -std=c11 -Wno-unused-variable -Wno-unknown-warning-option -Wno-unused-but-set-variable -Wno-unused-parameter -Wno-infinite-recursion -g -fwrapv -D_BSD_SOURCE -D_DEFAULT_SOURCE -fPIC -Wno-unused -Wno-parentheses -Wno-deprecated-declarations -Wno-#warnings -Wno-error=cpp -Wno-cpp -g -std=gnu11 -O3 -I "/home/opam/.opam/4.12/lib/ctypes" -I "/home/opam/.opam/4.12/lib/ocaml"   -c -o Hacl_Blake2b_32.o Hacl_Blake2b_32.c
# cc -fPIC -I. -I ../kremlin/include -I ../kremlin/kremlib/dist/minimal -Wall -Wextra -Werror -std=c11 -Wno-unused-variable -Wno-unknown-warning-option -Wno-unused-but-set-variable -Wno-unused-parameter -Wno-infinite-recursion -g -fwrapv -D_BSD_SOURCE -D_DEFAULT_SOURCE -fPIC -Wno-unused -Wno-parentheses -Wno-deprecated-declarations -Wno-#warnings -Wno-error=cpp -Wno-cpp -g -std=gnu11 -O3 -I "/home/opam/.opam/4.12/lib/ctypes" -I "/home/opam/.opam/4.12/lib/ocaml"   -c -o Hacl_Hash.o Hacl_Hash.c
# cc -fPIC -I. -I ../kremlin/include -I ../kremlin/kremlib/dist/minimal -Wall -Wextra -Werror -std=c11 -Wno-unused-variable -Wno-unknown-warning-option -Wno-unused-but-set-variable -Wno-unused-parameter -Wno-infinite-recursion -g -fwrapv -D_BSD_SOURCE -D_DEFAULT_SOURCE -fPIC -Wno-unused -Wno-parentheses -Wno-deprecated-declarations -Wno-#warnings -Wno-error=cpp -Wno-cpp -g -std=gnu11 -O3 -I "/home/opam/.opam/4.12/lib/ctypes" -I "/home/opam/.opam/4.12/lib/ocaml" -march=armv8-a+simd   -c -o Hacl_Blake2s_128.o Hacl_Blake2s_128.c
# In file included from Hacl_Blake2s_128.c:25:
# Hacl_Blake2s_128.h:48:3: error: unknown type name 'Lib_IntVector_Intrinsics_vec128'; did you mean 'Lib_IntVector_Intrinsics_vec256'?
#    Lib_IntVector_Intrinsics_vec128 *s,
#    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#    Lib_IntVector_Intrinsics_vec256
# Hacl_Blake2s_128.h:55:3: error: unknown type name 'Lib_IntVector_Intrinsics_vec128'; did you mean 'Lib_IntVector_Intrinsics_vec256'?
#    Lib_IntVector_Intrinsics_vec128 *s,
#    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#    Lib_IntVector_Intrinsics_vec256
# Hacl_Blake2s_128.h:63:3: error: unknown type name 'Lib_IntVector_Intrinsics_vec128'; did you mean 'Lib_IntVector_Intrinsics_vec256'?
#    Lib_IntVector_Intrinsics_vec128 *s,
#    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#    Lib_IntVector_Intrinsics_vec256
# Hacl_Blake2s_128.h:74:3: error: unknown type name 'Lib_IntVector_Intrinsics_vec128'; did you mean 'Lib_IntVector_Intrinsics_vec256'?
#    Lib_IntVector_Intrinsics_vec128 *wv,
#    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#    Lib_IntVector_Intrinsics_vec256
# Hacl_Blake2s_128.h:75:3: error: unknown type name 'Lib_IntVector_Intrinsics_vec128'; did you mean 'Lib_IntVector_Intrinsics_vec256'?
#    Lib_IntVector_Intrinsics_vec128 *hash,
#    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#    Lib_IntVector_Intrinsics_vec256
# Hacl_Blake2s_128.h:84:3: error: unknown type name 'Lib_IntVector_Intrinsics_vec128'; did you mean 'Lib_IntVector_Intrinsics_vec256'?
#    Lib_IntVector_Intrinsics_vec128 *wv,
#    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#    Lib_IntVector_Intrinsics_vec256
# Hacl_Blake2s_128.h:85:3: error: unknown type name 'Lib_IntVector_Intrinsics_vec128'; did you mean 'Lib_IntVector_Intrinsics_vec256'?
#    Lib_IntVector_Intrinsics_vec128 *hash,
#    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#    Lib_IntVector_Intrinsics_vec256
# Hacl_Blake2s_128.h:94:3: error: unknown type name 'Lib_IntVector_Intrinsics_vec128'; did you mean 'Lib_IntVector_Intrinsics_vec256'?
#    Lib_IntVector_Intrinsics_vec128 *wv,
#    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#    Lib_IntVector_Intrinsics_vec256
# Hacl_Blake2s_128.h:95:3: error: unknown type name 'Lib_IntVector_Intrinsics_vec128'; did you mean 'Lib_IntVector_Intrinsics_vec256'?
#    Lib_IntVector_Intrinsics_vec128 *hash,
#    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#    Lib_IntVector_Intrinsics_vec256
# Hacl_Blake2s_128.h:105:3: error: unknown type name 'Lib_IntVector_Intrinsics_vec128'; did you mean 'Lib_IntVector_Intrinsics_vec256'?
#    Lib_IntVector_Intrinsics_vec128 *hash
#    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#    Lib_IntVector_Intrinsics_vec256
# Hacl_Blake2s_128.c:28:20: error: unknown type name 'Lib_IntVector_Intrinsics_vec128'; did you mean 'Lib_IntVector_Intrinsics_vec256'?
#  update_blake2s_128(Lib_IntVector_Intrinsics_vec128 *s, uint64_t totlen, uint8_t *block)
#                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#                     Lib_IntVector_Intrinsics_vec256
# Hacl_Blake2s_128.c:58:90: error: expected end of line before 'ignored'
#      KRML_CHECK_SIZE(sizeof (Lib_IntVector_Intrinsics_vec128), (uint32_t)4U * (uint32_t)1U);
#                                                                                           ^
# Hacl_Blake2s_128.c:204:3: error: unknown type name 'Lib_IntVector_Intrinsics_vec128'; did you mean 'Lib_IntVector_Intrinsics_vec256'?
#    Lib_IntVector_Intrinsics_vec128 *s,
#    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#    Lib_IntVector_Intrinsics_vec256
# Hacl_Blake2s_128.c:210:47: error: expected end of line before 'ignored'
#    KRML_CHECK_SIZE(sizeof (uint8_t), double_row);
#                                                ^
# Hacl_Blake2s_128.c:226:3: error: unknown type name 'Lib_IntVector_Intrinsics_vec128'; did you mean 'Lib_IntVector_Intrinsics_vec256'?
#    Lib_IntVector_Intrinsics_vec128 *s,
#    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#    Lib_IntVector_Intrinsics_vec256
# Hacl_Blake2s_128.c:243:3: error: unknown type name 'Lib_IntVector_Intrinsics_vec128'; did you mean 'Lib_IntVector_Intrinsics_vec256'?
#    Lib_IntVector_Intrinsics_vec128 *s,
#    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#    Lib_IntVector_Intrinsics_vec256
# Hacl_Blake2s_128.c:280:88: error: expected end of line before 'ignored'
#    KRML_CHECK_SIZE(sizeof (Lib_IntVector_Intrinsics_vec128), (uint32_t)4U * (uint32_t)1U);
#                                                                                         ^
# Hacl_Blake2s_128.c:312:90: error: expected end of line before 'ignored'
#      KRML_CHECK_SIZE(sizeof (Lib_IntVector_Intrinsics_vec128), (uint32_t)4U * (uint32_t)1U);
#                                                                                           ^
# Hacl_Blake2s_128.c:463:3: error: unknown type name 'Lib_IntVector_Intrinsics_vec128'; did you mean 'Lib_IntVector_Intrinsics_vec256'?
#    Lib_IntVector_Intrinsics_vec128 *wv,
#    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#    Lib_IntVector_Intrinsics_vec256
# Hacl_Blake2s_128.c:464:3: error: unknown type name 'Lib_IntVector_Intrinsics_vec128'; did you mean 'Lib_IntVector_Intrinsics_vec256'?
#    Lib_IntVector_Intrinsics_vec128 *hash,
#    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#    Lib_IntVector_Intrinsics_vec256
# Hacl_Blake2s_128.c:502:90: error: expected end of line before 'ignored'
#      KRML_CHECK_SIZE(sizeof (Lib_IntVector_Intrinsics_vec128), (uint32_t)4U * (uint32_t)1U);
#                                                                                           ^
# Hacl_Blake2s_128.c:647:3: error: unknown type name 'Lib_IntVector_Intrinsics_vec128'; did you mean 'Lib_IntVector_Intrinsics_vec256'?
#    Lib_IntVector_Intrinsics_vec128 *wv,
#    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#    Lib_IntVector_Intrinsics_vec256
# Hacl_Blake2s_128.c:648:3: error: unknown type name 'Lib_IntVector_Intrinsics_vec128'; did you mean 'Lib_IntVector_Intrinsics_vec256'?
#    Lib_IntVector_Intrinsics_vec128 *hash,
#    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#    Lib_IntVector_Intrinsics_vec256
# Hacl_Blake2s_128.c:686:3: error: unknown type name 'Lib_IntVector_Intrinsics_vec128'; did you mean 'Lib_IntVector_Intrinsics_vec256'?
#    Lib_IntVector_Intrinsics_vec128 *wv,
#    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#    Lib_IntVector_Intrinsics_vec256
# Hacl_Blake2s_128.c:687:3: error: unknown type name 'Lib_IntVector_Intrinsics_vec128'; did you mean 'Lib_IntVector_Intrinsics_vec256'?
#    Lib_IntVector_Intrinsics_vec128 *hash,
#    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#    Lib_IntVector_Intrinsics_vec256
# Hacl_Blake2s_128.c:704:3: error: unknown type name 'Lib_IntVector_Intrinsics_vec128'; did you mean 'Lib_IntVector_Intrinsics_vec256'?
#    Lib_IntVector_Intrinsics_vec128 *wv,
#    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#    Lib_IntVector_Intrinsics_vec256
# Hacl_Blake2s_128.c:705:3: error: unknown type name 'Lib_IntVector_Intrinsics_vec128'; did you mean 'Lib_IntVector_Intrinsics_vec256'?
#    Lib_IntVector_Intrinsics_vec128 *hash,
#    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#    Lib_IntVector_Intrinsics_vec256
# Hacl_Blake2s_128.c:722:3: error: unknown type name 'Lib_IntVector_Intrinsics_vec128'; did you mean 'Lib_IntVector_Intrinsics_vec256'?
#    Lib_IntVector_Intrinsics_vec128 *wv,
#    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#    Lib_IntVector_Intrinsics_vec256
# Hacl_Blake2s_128.c:723:3: error: unknown type name 'Lib_IntVector_Intrinsics_vec128'; did you mean 'Lib_IntVector_Intrinsics_vec256'?
#    Lib_IntVector_Intrinsics_vec128 *hash,
#    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#    Lib_IntVector_Intrinsics_vec256
# Hacl_Blake2s_128.c:751:3: error: unknown type name 'Lib_IntVector_Intrinsics_vec128'; did you mean 'Lib_IntVector_Intrinsics_vec256'?
#    Lib_IntVector_Intrinsics_vec128 *hash
#    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#    Lib_IntVector_Intrinsics_vec256
# Hacl_Blake2s_128.c:755:47: error: expected end of line before 'ignored'
#    KRML_CHECK_SIZE(sizeof (uint8_t), double_row);
#                                                ^
# Hacl_Blake2s_128.c: In function 'Hacl_Blake2s_128_blake2s':
# Hacl_Blake2s_128.c:780:3: error: unknown type name 'Lib_IntVector_Intrinsics_vec128'; did you mean 'Lib_IntVector_Intrinsics_vec256'?
#    Lib_IntVector_Intrinsics_vec128 stzero = Lib_IntVector_Intrinsics_vec128_zero;
#    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#    Lib_IntVector_Intrinsics_vec256
# Hacl_Blake2s_128.c:780:44: error: 'Lib_IntVector_Intrinsics_vec128_zero' undeclared (first use in this function); did you mean 'Lib_IntVector_Intrinsics_vec256'?
#    Lib_IntVector_Intrinsics_vec128 stzero = Lib_IntVector_Intrinsics_vec128_zero;
#                                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#                                             Lib_IntVector_Intrinsics_vec256
# Hacl_Blake2s_128.c:780:44: note: each undeclared identifier is reported only once for each function it appears in
# In file included from ../kremlin/kremlib/dist/minimal/FStar_UInt128_Verified.h:12,
#                  from ../kremlin/include/kremlin/internal/types.h:101,
#                  from Hacl_Blake2s_128.h:34,
#                  from Hacl_Blake2s_128.c:25:
# Hacl_Blake2s_128.c:781:27: error: 'Lib_IntVector_Intrinsics_vec128' undeclared (first use in this function); did you mean 'Lib_IntVector_Intrinsics_vec256'?
#    KRML_CHECK_SIZE(sizeof (Lib_IntVector_Intrinsics_vec128), stlen);
#                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# ../kremlin/include/kremlin/internal/target.h:89:48: note: in definition of macro 'KRML_CHECK_SIZE'
#      if (((size_t)(sz)) > ((size_t)(SIZE_MAX / (size_elt)))) {                  \
#                                                 ^~~~~~~~
# Hacl_Blake2s_128.c:782:34: error: expected ';' before 'b'
#    Lib_IntVector_Intrinsics_vec128 b[stlen];
#                                   ^~
#                                   ;
# Hacl_Blake2s_128.c:784:5: error: 'b' undeclared (first use in this function)
#      b[_i] = stzero;
#      ^
# Hacl_Blake2s_128.c:795:34: error: expected ';' before 'b1'
#    Lib_IntVector_Intrinsics_vec128 b1[stlen];
#                                   ^~~
#                                   ;
# Hacl_Blake2s_128.c:797:5: error: 'b1' undeclared (first use in this function)
#      b1[_i] = stzero;
#      ^~
# Hacl_Blake2s_128.c:798:3: error: implicit declaration of function 'Hacl_Blake2s_128_blake2s_init'; did you mean 'Hacl_Blake2s_32_blake2s_init'? [-Werror=implicit-function-declaration]
#    Hacl_Blake2s_128_blake2s_init(b1, b, kk, k, nn);
#    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#    Hacl_Blake2s_32_blake2s_init
# Hacl_Blake2s_128.c:799:3: error: implicit declaration of function 'blake2s_update_blocks' [-Werror=implicit-function-declaration]
#    blake2s_update_blocks(ll, b1, b, prev0, d);
#    ^~~~~~~~~~~~~~~~~~~~~
# Hacl_Blake2s_128.c:800:3: error: implicit declaration of function 'Hacl_Blake2s_128_blake2s_finish'; did you mean 'Hacl_Blake2s_32_blake2s_finish'? [-Werror=implicit-function-declaration]
#    Hacl_Blake2s_128_blake2s_finish(nn, output, b);
#    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#    Hacl_Blake2s_32_blake2s_finish
# Hacl_Blake2s_128.c: At top level:
# cc1: error: unrecognized command line option '-Wno-#warnings' [-Werror]
# cc1: error: unrecognized command line option '-Wno-infinite-recursion' [-Werror]
# cc1: error: unrecognized command line option '-Wno-unknown-warning-option' [-Werror]
# cc1: all warnings being treated as errors
# make: *** [<builtin>: Hacl_Blake2s_128.o] Error 1
# make: Leaving directory '/home/opam/.opam/4.12/.opam-switch/build/hacl-star-raw.0.3.2/raw'
```

Reply:
Seen on: https://github.com/ocaml/opam-repository/pull/19065